connect/runtime/src/main/java/org/apache/kafka/connect/util/TopicAdmin.java

```diff
@@ -0,0 +1,368 @@
+    }
+
+    /**
+     * Given the topic definition, check whether the topic exists and if not then attempt to create the
+     * missing topic. The brokers will use its default topic options for anything not specified in the supplied definition.
+     * This method returns the topic details for the newly-created or previously-existing topic.
+     * <p>
+     * This operation requires the broker to support clients performing several administrative operations. If the broker
+     * does not support this capability, this method returns null.
+     *
+     * @param topic the specifications of the topic
+     * @return the topic description of the newly-created or existing topic; null only
+     * if the broker is too old to perform this operation
+     * @throws ConnectException if an error occurs, the operation takes too long, or the thread is interrupted while
+     *                          attempting to perform this operation
+     */
+    public TopicDescription createTopicIfMissing(NewTopic topic) {
+        Map<String, TopicDescription> result = createTopicsIfMissing(topic);
+        return result != null ? result.get(topic.name()) : null;
+    }
+
+    /**
+     * Given the topic definitions, check whether the topics exists and for all those that do not attempt to create the
+     * missing topics. The brokers will use its default topic options for anything not specified in the supplied definitions.
+     * This method returns the topic details for all of the topic included in the supplied descriptions.
+     * <p>
+     * This operation requires the broker to support clients performing several administrative operations. If the broker
+     * does not support this capability, this method returns null.
+     *
+     * @param topics the specifications of the topics
+     * @return the topic description of the newly-created or existing topics, keyed by the topic names; null only
+     * if the broker is too old to perform this operation
+     * @throws ConnectException if an error occurs, the operation takes too long, or the thread is interrupted while
+     *                          attempting to perform this operation
+     */
+    public Map<String, TopicDescription> createTopicsIfMissing(NewTopic... topics) {
+        if (topics == null || topics.length == 0) return Collections.emptyMap();
+
+        Map<String, NewTopic> topicsByName = new HashMap<>();
+        for (NewTopic topic : topics) {
+            if (topic != null) topicsByName.put(topic.name(), topic);
+        }
+        Set<String> allRequestedTopicNames = new HashSet<>(topicsByName.keySet());
+        String bootstrapServers = bootstrapServers();
+
+        try {
+            // Getting topic descriptions for non-existant topics may auto-create them, so first see which of
+            // our topics do already exist
+            Set<String> existingTopicNames = existingTopics(topicsByName.keySet());
+            if (existingTopicNames == null) {
```

Comment:
All the `null` checks at each layer of the call stack make me think that particular issue might be better handled with an exception.

Not critical since this is all internal code, but seems like then we'd only need to check version compatibility in one or two places.